src/app/clusters/commodity-tariff-server/CommodityTariffAttrsDataMgmt.h

```diff
@@ -0,0 +1,942 @@
+        {
+            return nullptr;
+        }
+
+        if constexpr (IsValueNullable())
+        {
+            if (GetValueRef().IsNull())
+            {
+                return nullptr;
+            }
+            if constexpr (IsList<WrappedType>::value)
+            {
+                return GetValueRef().Value().data();
+            }
+            else
+            {
+                return &GetValueRef().Value();
+            }
+        }
+        else if constexpr (IsValueList())
+        {
+            return GetValueRef().data();
+        }
+        else
+        {
+            return &GetValueRef();
+        }
+    }
+
+    /**
+     * @brief Check if current update is validated
+     * @return true if in kValidated state
+     */
+    bool IsValid() const { return (mUpdateState == UpdateState::kValidated); }
+
+    /**
+     * @brief Check if value storage contains valid data
+     * @return true if storage is in kHold state
+     */
+    bool HasValue() const { return (mHoldState == StorageState::kHold); }
+
+    /**
+     * @brief Prepares a new value for modification
+     * @param size Number of elements to allocate (for list types only)
+     * @return CHIP_NO_ERROR on success, or:
+     *         - CHIP_ERROR_NO_MEMORY if allocation fails
+     *         - CHIP_ERROR_INVALID_LIST_LENGTH if size < 1 for list types
+     *         - CHIP_ERROR_INCORRECT_STATE if called during active update
+     *
+     * @note For list types:
+     *       - Allocates memory for specified number of elements
+     *       - Initializes list storage
+     *       - Transitions state to kInitialized
+     * @note For non-list types:
+     *       - Creates default-initialized value
+     *       - Ignores size parameter
+     */
+    CHIP_ERROR CreateNewValue(size_t size)
+    {
+        if (mUpdateState != UpdateState::kIdle)
+        {
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+
+        if constexpr (IsValueList() || IsList<WrappedType>::value)
+        {
+            if (size >= 1)
+            {
+                auto * buffer = static_cast<PayloadType *>(chip::Platform::MemoryCalloc(size, sizeof(PayloadType)));
+
+                if (!buffer)
+                {
+                    return CHIP_ERROR_NO_MEMORY;
+                }
+
+                if constexpr (IsValueNullable())
+                {
+                    GetNewValueRef().SetNonNull(DataModel::List<PayloadType>(buffer, size));
+                }
+                else
+                {
+                    GetNewValueRef() = DataModel::List<PayloadType>(buffer, size);
+                }
+            }
+            else
+            {
+                if constexpr (IsValueNullable())
+                {
+                    GetNewValueRef().SetNull();
+                }
+            }
+        }
+        else if constexpr (IsValueNullable())
+        {
+            WrappedType tmp = WrappedType();
+            GetNewValueRef().SetNonNull(tmp);
```

Comment:
Why are we defaulting to non-null here?  Should be documented.  Probably something to do with being able to call GetNewValueData?